src/http/api/resources/files-regular.js

```diff
@@ -11,16 +11,20 @@ const Boom = require('@hapi/boom')
 const { PassThrough } = require('stream')
 const multibase = require('multibase')
 const isIpfs = require('is-ipfs')
-const { promisify } = require('util')
 const { cidToString } = require('../../../utils/cid')
 const { Format } = require('../../../core/components/refs')
 const pipe = require('it-pipe')
 const all = require('it-all')
-const concat = require('it-concat')
 const ndjson = require('iterable-ndjson')
 const { map } = require('streaming-iterables')
 const streamResponse = require('../../utils/stream-response')
 
+const toBuffer = async function * (source) {
+  for await (const chunk of source) {
+    yield chunk.slice()
+  }
+}
```

Comment:
this pattern is begging for a `it-transform`

Reply:
Could use this: https://www.npmjs.com/package/it-buffer, it's already used in libp2p.